src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -549,23 +546,24 @@ private async Task ConnectFirstAsync(CancellationToken cancellationToken)
                 }
                 catch (Exception e) // gather exceptions
                 {
-                    exceptions ??= new List<Exception>();
-                    exceptions.Add(e);
+                    _logger.LogError(e, "Connection attempt has thrown.");
                     canRetry = false; // retry strategy threw
+                    isExceptionThrown = true;
                 }
 
             } while (canRetry);
 
-            var aggregate = exceptions == null ? null : new AggregateException(exceptions);
+            string msgSomeThingWentWrong = "";
+
+            if (isExceptionThrown)
+                msgSomeThingWentWrong = "Exception(s) is thrown in the past. Please, see the logs.";
```

Comment:
"Some exceptions were thrown and have been written to the log. Please refer to the log for details."